source/common/router/header_parser.h

```diff
@@ -70,6 +70,7 @@ class HeaderParser : public Http::HeaderEvaluator {
   struct HeadersToAddEntry {
     HeaderFormatterPtr formatter_;
     const std::string original_value_;
+    const bool add_if_empty_ = false;
```

Comment:
If this is const and initialized here, should it be constexpr?

Reply:
Non-static data members can't be constexpr.

Reply:
I might be missing something here, but isn't this always false (i.e., initialized here, and doesn't change)?
If so, why is it per-entry data-member?

Reply:
false is just the default value, it's still possible to set the value using the default constructor -- i.e. HeadersToAddEntry{std::move(formatter), "orig_val", true};

Reply:
Ah, thanks. I've missed the struct initializer.